packages/collection/src/index.ts

```diff
@@ -9,6 +9,12 @@ export interface CollectionConstructor {
 	readonly [Symbol.species]: CollectionConstructor;
 }
 
+/**
+ * Represents an immutable version of a collection
+ */
+export type ReadonlyCollection<K, V> = ReadonlyMap<K, V> &
+	Omit<Collection<K, V>, 'forEach' | 'ensure' | 'sweep' | 'sort'>;
```

Comment:
Won't this pick set, delete, etc? Also, why are you omitting forEach?

Reply:
I just fixed the get/set/delete. However `forEach` is omitted because ReadonlyMap uses its own signature that only accepts readonly maps. Since collection simply inherits from map I omitted from the collection and used `ReadonlyMap`s foreach instead.